library/std/src/os/horizon/raw.rs

```diff
@@ -9,6 +9,7 @@
             definitions"
 )]
 #![allow(deprecated)]
+#![allow(dead_code)]
```

Comment:
What/when does this flag? It seems like either it should apply to only a couple objects, or the whole module should be disabled in some cases

Reply:
Fair point, actually revisiting this I am surprised it warned at all:
```
error: struct `stat` is never constructed
  --> library/std/src/os/horizon/raw.rs:41:12
   |
41 | pub struct stat {
   |            ^^^^
   |
   = note: `stat` has a derived impl for the trait `Clone`, but this is intentionally ignored during dead code analysis
   = note: `-D dead-code` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(dead_code)]`

warning: `std` (lib) generated 1 warning
```

Shouldn't `pub struct` normally avoid emitting a `dead_code` warning?

I can move the attribute to just that struct, but is this actually some kind of compiler regression or am I missing something?

Reply:
> Shouldn't `pub struct` normally avoid emitting a `dead_code` warning?

@ian-h-chamberlain The `raw` module is only exported at a crate-level. As such, if never built, the compiler will complain about unused members (such as the `stat` struct) because they aren't actually publicly accessible API.

Reply:
Oh, I see - most `os` crates have their `raw` module public, Horizon is special in `pub(crate)` since this module was added after `std::os::raw` was already deprecated. So I guess it's just around for consistency, per https://github.com/rust-lang/rust/pull/95897#discussion_r889686808.

So this seems fine to me, but could you add a note explaining this? Or just update the crate's docstring.

Has this just been a warning for a while then? Assuming nothing in this PR is making it suddenly start to fire.

Reply:
Yes, I think this warning has been present since the module was added — at the time, there were a lot of other warnings when building for this target, but they've mostly been cleaned up now and this was the last remaining warning.

I've moved the `#[allow]` to be more specific and added a comment.

Reply:
Thanks for the change!